src/osemgrep/networking/Rule_fetching.mli

```diff
@@ -76,7 +78,7 @@ val rules_from_dashdash_config_async :
   registry_caching:bool ->
   < Cap.network > ->
   Rules_config.t ->
-  rules_and_origin list Lwt.t
+  (rules_and_origin list * Rule.error list) Lwt.t
```

Comment:
maybe better to use a result type here? if the error list is not empty, then rules_and_origin must be empty right?
In that case better to encode that in the type no?

Reply:
`rules_from_dashdash_config` actually can have both `rules_and_origin` and `errors`, because `--config` can potentially be given multiple configs (such as from a directory). Some of these might succeed to parse, some might not.